addons/stock/models/product.py

```diff
@@ -173,10 +173,11 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
         quants_res = dict((item['product_id'][0], (item['quantity'], item['reserved_quantity'])) for item in Quant._read_group(domain_quant, ['product_id', 'quantity', 'reserved_quantity'], ['product_id'], orderby='id'))
         if dates_in_the_past:
             # Calculate the moves that were done before now to calculate back in time (as most questions will be recent ones)
-            domain_move_in_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_in_done
-            domain_move_out_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_out_done
-            moves_in_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_in_done, ['product_id', 'product_qty'], ['product_id'], orderby='id'))
-            moves_out_res_past = dict((item['product_id'][0], item['product_qty']) for item in Move._read_group(domain_move_out_done, ['product_id', 'product_qty'], ['product_id'], orderby='id'))
+            MoveLine = self.env['stock.move.line'].with_context(active_test=False)
+            domain_move_line_in_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_in_done
+            domain_move_line_out_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_out_done
+            move_lines_in_res_past = dict((item['product_id'][0], item['qty_done']) for item in MoveLine._read_group(domain_move_line_in_done, ['product_id', 'qty_done'], ['product_id'], orderby='id'))
```

Comment:
Hello, you are right on this. But there is still a potential issue with your patch. The `qty_done` field on stock.move.line is expressed in the SML UoM, not the product one. It was not the case for `product_qty` of stock.move as it is a computed field to have the product's UoM quantity.

You should maybe add `product_uom_id` in the grouping fields of `read_group`, convert the quantities, then sum them all.

Reply:
Hi, i added a compute field done_product_qty like on stock.move